internal/services/dataprotection/azurerm_data_protection_backup_policy_kubernetes_cluster_resource_test.go

```diff
@@ -0,0 +1,250 @@
+  name                = "acctest-aks-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  vault_name          = azurerm_data_protection_backup_vault.test.name
+
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+
+  retention_rule {
+    name     = "Daily"
+    priority = 25
+
+    life_cycle {
+      duration        = "P84D"
+      data_store_type = "OperationalStore"
+    }
+
+    criteria {
+      days_of_week           = ["Thursday"]
+      months_of_year         = ["November"]
+      weeks_of_month         = ["First"]
+      scheduled_backup_times = ["2021-05-23T02:30:00Z"]
+    }
+  }
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P7D"
+      data_store_type = "OperationalStore"
+    }
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func (r DataProtectionBackupPolicyKubernatesClusterTestResource) requiresImport(data acceptance.TestData) string {
+	config := r.basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_kubernetes_cluster" "import" {
+  name                = azurerm_data_protection_backup_policy_kubernetes_cluster.test.name
+  resource_group_name = azurerm_data_protection_backup_policy_kubernetes_cluster.test.resource_group_name
+  vault_name          = azurerm_data_protection_backup_policy_kubernetes_cluster.test.vault_name
+
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+
+  retention_rule {
+    name     = "Daily"
+    priority = 25
+
+    life_cycle {
+      duration        = "P84D"
+      data_store_type = "OperationalStore"
+    }
+
+    criteria {
+      days_of_week           = ["Thursday"]
+      months_of_year         = ["November"]
+      scheduled_backup_times = ["2021-05-23T02:30:00Z"]
+    }
+  }
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P7D"
+      data_store_type = "OperationalStore"
+    }
+  }
+}
+`, config)
+}
+
+func (r DataProtectionBackupPolicyKubernatesClusterTestResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_kubernetes_cluster" "test" {
+  name                            = "acctest-aks-%d"
+  resource_group_name             = azurerm_resource_group.test.name
+  vault_name                      = azurerm_data_protection_backup_vault.test.name
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+  time_zone                       = "India Standard Time"
+
+  retention_rule {
+    name     = "Daily"
+    priority = 25
+
+    life_cycle {
+      duration        = "P7D"
+      data_store_type = "OperationalStore"
+      target_copy_setting {
+        copy_after = jsonencode({
+          objectType = "ImmediateCopyOption"
+        })
+        data_store_type = "OperationalStore"
+      }
+    }
+
+    life_cycle {
+      duration        = "P84D"
+      data_store_type = "OperationalStore"
+    }
```

Comment:
does it make sense to have 2 `life_cycle` blocks with the same `data_store_type`?

Reply:
Yes, the `duration` may vary. Below is the real data created in Azure Portal.
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/aaec6f76-00f5-4d26-b8f3-b28db4c86bbf)